sysroot_creator/scripts/create_cc_sysroot.py

```diff
@@ -142,184 +147,160 @@ def __str__(self):
-    return cc_system_setup_script_path
-
-
-def setup_sysroot_environment(
-        sys_setup_path: Path, build_setup_path: Path) -> None:
-    """Set up the environment with variables and symbolic links."""
-    logger.info('Sourcing sysroot environment...')
-    logger.info("Executing 'bash {}'".format(sys_setup_path))
-    subprocess.run(['bash', str(sys_setup_path)])
-    logger.info("Executing 'source {}'".format(build_setup_path))
-    subprocess.run(['source', str(build_setup_path)], shell=True)
+class SysrootCreator:
+    def __init__(
+      self,
+      cc_root_dir: str,
+      ros_workspace_dir: str,
+      platform: Platform,
+      docker_config: DockerConfig):
+        self._cc_root_dir = Path(cc_root_dir)
+        self._ros_workspace_dir = Path(ros_workspace_dir)
+        self._target_sysroot = self._cc_root_dir / SYSROOT_DIR_NAME
+        self._ros_ws_directory = self._target_sysroot / self._ros_workspace_dir
+        self._qemu_directory = self._target_sysroot / QEMU_DIR_NAME
+        self._dockerfile_directory = Path(__file__).parent / DOCKER_WS_NAME
+        self._expected_dockerfile_directory = self._target_sysroot / DOCKER_WS_NAME
+        self.system_setup_script_path = Path()
+        self.build_setup_script_path = Path()
+        self._platform = platform
+        self._docker_config = docker_config
+
+        try:
+            self._setup_sysroot_dir()
+        except FileNotFoundError as e:
+            print(e)
+            sys.exit(1)
+
+    def _setup_sysroot_dir(self) -> None:
+        logger.info('Checking sysroot directory...')
+        if self._target_sysroot.exists():
+            logger.debug('Sysroot directory exists.')
+            if not self._ros_ws_directory.exists():
+                raise FileNotFoundError(ROS_WS_DIR_ERROR_STRING.substitute(
+                    ros_ws=self._ros_workspace_dir))
+            logger.debug('ROS workspace exists.')
+            if not self._qemu_directory.exists():
+                raise FileNotFoundError(QEMU_DIR_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
+            if not os.listdir(str(self._qemu_directory.absolute())):
+                raise FileNotFoundError(QEMU_EMPTY_ERROR_STRING.substitute(qemu_dir=QEMU_DIR_NAME))
+            logger.debug('QEMU binaries exist')
+            shutil.copy(str(self._dockerfile_directory), str(self._target_sysroot))
+            if not self._expected_dockerfile_directory.exists():
+                raise FileNotFoundError(COPY_DOCKER_WS_ERROR_STRING.substitute(
+                    dockerfile=DOCKER_WS_NAME))
+        else:
+            raise FileNotFoundError(SYSROOT_NOT_FOUND_ERROR_STRING.substitute(
+                sysroot_dir=self._cc_root_dir))
+
+    def build_workspace_sysroot_image(self) -> None:
+        """Build the target sysroot docker image."""
+        logger.info('Fetching sysroot base image: {}'.format(self._docker_config.base_image))
+        DOCKER_CLIENT.images.pull(self._docker_config.base_image)
```

Comment:
would be nice to make it `self._docker_client` down the road so that we can mock it easily.